inference-engine/src/transformations/src/transformations/op_conversions/simplify_ctc_greedy_decoder_seq_len.cpp

```diff
@@ -0,0 +1,111 @@
+                decoder_seq_len->input_value(1), mask_shape->output(0));
+        auto transpose_upper_bounds = std::make_shared<ngraph::opset6::Transpose>(upper_bounds->output(0),
+                                                                                  ngraph::opset6::Constant::create(seq_len_type,
+                                                                                                                   Shape({2}), {1, 0}));
+        // Compute boolean sequence mask
+        auto bool_seq_mask = std::make_shared<ngraph::opset6::GreaterEqual>(transpose_upper_bounds->output(0),
+                                                                            range1T->output(0));
+
+        // Generate resulted seq mask
+        auto mask_val_true = ngraph::opset6::Constant::create(seq_len_type, Shape{}, {1});
+        auto mask_val_false = ngraph::opset6::Constant::create(seq_len_type, Shape{}, {0});
+        auto seq_mask = std::make_shared<ngraph::opset6::Select>(bool_seq_mask, mask_val_true, mask_val_false);
+        auto transpose_seq_mask = std::make_shared<ngraph::opset6::Transpose>(seq_mask->output(0),
+                                                                              ngraph::opset6::Constant::create(seq_len_type,
+                                                                                                               Shape({2}), {1, 0}));
+        auto transpose_seq_mask_f32 = std::make_shared<ngraph::opset6::Convert>(transpose_seq_mask->output(0), element::f32);
+        // Create CTCGreedyDecoder with original merge_repeated attribute and connect data and resulted seq_mask
+        auto decoder = std::make_shared<ngraph::opset6::CTCGreedyDecoder>(transpose,
+                                                                                     transpose_seq_mask_f32->output(0),
+                                                                                     decoder_seq_len->get_merge_repeated());
+        decoder->set_friendly_name(decoder_seq_len->get_friendly_name());
+
+        // Normalize output from CTCGreedyDecoder = output_f and create second output with output_seq_len
+        auto squeeze2_axis = ngraph::opset6::Constant::create(seq_len_type, Shape({1}), {3});
+        auto squeeze2_output_f = std::make_shared<ngraph::opset6::Squeeze>(decoder->output(0), squeeze2_axis);
+        auto squeeze1_axis = ngraph::opset6::Constant::create(seq_len_type, Shape({1}), {2});
+        auto squeeze1_output_f = std::make_shared<ngraph::opset6::Squeeze>(squeeze2_output_f->output(0), squeeze1_axis);
+
+        element::Type ci_type = decoder_seq_len->get_classes_index_type();
+        element::Type sl_type = decoder_seq_len->get_sequence_length_type();
+        // Receive the first output with correct classes_index_type
+        auto output_i = std::make_shared<ngraph::opset6::Convert>(squeeze1_output_f->output(0), ci_type);
+        auto minus1 = opset6::Constant::create(ci_type, Shape{}, {-1});
```

Comment:
Is it possible that `ci_type` can be the same as `squeeze1_output_f->output(0)` type?

Reply:
No. According to spec CTCGreedyDecoder v1 return float output. We need to call convert for change type to ci_type.